community/jmx/src/test/java/org/neo4j/jmx/impl/StoreSizeBeanTest.java

```diff
@@ -0,0 +1,271 @@
+import static org.neo4j.kernel.impl.store.StoreFile.NODE_LABEL_STORE;
+import static org.neo4j.kernel.impl.store.StoreFile.NODE_STORE;
+import static org.neo4j.kernel.impl.store.StoreFile.PROPERTY_ARRAY_STORE;
+import static org.neo4j.kernel.impl.store.StoreFile.PROPERTY_KEY_TOKEN_NAMES_STORE;
+import static org.neo4j.kernel.impl.store.StoreFile.PROPERTY_KEY_TOKEN_STORE;
+import static org.neo4j.kernel.impl.store.StoreFile.PROPERTY_STORE;
+import static org.neo4j.kernel.impl.store.StoreFile.PROPERTY_STRING_STORE;
+import static org.neo4j.kernel.impl.store.StoreFile.RELATIONSHIP_GROUP_STORE;
+import static org.neo4j.kernel.impl.store.StoreFile.RELATIONSHIP_STORE;
+import static org.neo4j.kernel.impl.store.StoreFile.RELATIONSHIP_TYPE_TOKEN_NAMES_STORE;
+import static org.neo4j.kernel.impl.store.StoreFile.RELATIONSHIP_TYPE_TOKEN_STORE;
+import static org.neo4j.kernel.impl.store.StoreFile.SCHEMA_STORE;
+import static org.neo4j.kernel.impl.storemigration.StoreFileType.ID;
+import static org.neo4j.kernel.impl.storemigration.StoreFileType.STORE;
+
+public class StoreSizeBeanTest
+{
+    private FileSystemAbstraction fs = new EphemeralFileSystemAbstraction();
+    private File storeDir = new File( "" );
+    private PhysicalLogFiles physicalLogFiles = new PhysicalLogFiles( storeDir, fs );
+    private LegacyIndexProviderLookup legacyIndexProviderLookup = mock( LegacyIndexProviderLookup.class );
+    private SchemaIndexProvider schemaIndexProvider = mock( SchemaIndexProvider.class );
+    private LabelScanStore labelScanStore = mock( LabelScanStore.class );
+    private StoreSize storeSizeBean;
+    private File storeDirAbsolute;
+
+    @Before
+    public void setUp() throws Throwable
+    {
+        DataSourceManager dataSourceManager = new DataSourceManager();
+        GraphDatabaseAPI db = mock( GraphDatabaseAPI.class );
+        NeoStoreDataSource dataSource = mock( NeoStoreDataSource.class );
+
+        // Setup all dependencies
+        Dependencies dependencies = new Dependencies();
+        dependencies.satisfyDependency( fs );
+        dependencies.satisfyDependencies( dataSourceManager );
+        dependencies.satisfyDependency( physicalLogFiles );
+        dependencies.satisfyDependency( legacyIndexProviderLookup );
+        dependencies.satisfyDependency( schemaIndexProvider );
+        dependencies.satisfyDependency( labelScanStore );
+        when( db.getDependencyResolver() ).thenReturn( dependencies );
+        when( dataSource.getDependencyResolver() ).thenReturn( dependencies );
+
+        // Start DataSourceManager
+        dataSourceManager.register( dataSource );
+        dataSourceManager.start();
+
+        // Create bean
+        KernelData kernelData = new DefaultKernelData( fs, mock( PageCache.class ), storeDir, Config.empty(), db );
+        ManagementData data = new ManagementData( new StoreSizeBean(), kernelData, ManagementSupport.load() );
+        storeSizeBean = (StoreSize) new StoreSizeBean().createMBean( data );
+    }
```

Comment:
While it would be better to have a less artificial scenario it's hard to deceive a reproducible test case that would catch any meaningful errors. The current test basically verifies the grouping of the files and have been renamed to reflect this.